python/ray/data/tests/test_stats.py

```diff
@@ -262,6 +339,127 @@ def test_dataset_pipeline_stats_basic(ray_start_regular_shared):
     )
 
 
+@patch("ray.data._internal.plan.logger")
+def test_dataset_pipeline_stats_autolog(mock_logger, ray_start_regular_shared):
+    """Tests functionality to autolog stats after each stage execution,
+    enabled via the `DatasetContext.enable_auto_log_stats` parameter."""
+    context = DatasetContext.get_current()
+    enable_auto_log_stats_orig = context.enable_auto_log_stats
+    optimize_fuse_stages_orig = context.optimize_fuse_stages
```

Comment:
wondering why you need to override `optimize_fuse_stages` to True? It sets to True by default - https://github.com/ray-project/ray/blob/master/python/ray/data/context.py#L37 .

Reply:
ah sorry, i was using [test_dataset_pipeline_stats_basic](https://github.com/ray-project/ray/blob/aa7ec7f21fc33510b9320d2688deeb81574f2107/python/ray/data/tests/test_stats.py#L274) as reference, which explicitly sets it to True. i can go ahead and remove this